lib/engine-addon.js

```diff
@@ -281,6 +282,15 @@ var buildEngineStyleTree = memoize(function buildEngineStyleTree() {
   return this.debugTree(combinedEngineStylesAndDependencyStylesTree, 'engine-style:input');
 });
 
+/**
+ * Generates an identifier for an addon, which will be used when
+ * addons are deduped. Different instances of the same addon will
+ * be detected based on this identifier.
+ */
+function generateAddonId(addon) {
```

Comment:
Why do we need to call this `addonId` instead of just using `addon.name`? The function isn't doing anything special, and by introducing a new concept for identifying addons, it makes the code much harder to reason about.